lib/rspec/core/lookup_set.rb

```diff
@@ -0,0 +1,32 @@
+module RSpec
+  module Core
+    # @private
+    class LookupSet
```

Comment:
> I didn't want to shadow the standard lib name especially as it's not functionally complete

What problem would that case, though?  User specs won't use this class when referencing `Set` unless they open the `RSpec::Core` module.  We've already shadowed a core class with `RSpec::Core::Time` and I've never seen it cause a problem...